Add prettier pre-commit hook

yakkomajuri opened the pull request:
When the user runs `git commit` this will trigger prettier in "write" mode so it will automatically lint the files according to the prettier configuration the repo already has at `.prettierrc`.

Also moved prettier to `devDependencies` but feel free to disagree with me there.

Close #380

maciekgrzybek reviewed (approved):
Looks good to me. @imbhargav5 ?

imbhargav5 commented:
Thanks a ton! @yakkomajuri 

Yeah `devDependencies` normally make sense but in yarn workspaces they aren't very different from `root` `dependencies`. So let's have them in `dependencies` if possible. 

Also, we have some automated commits that occur in GitHub actions, can we disable `lint-staged` to fire on such commits or is it already taken care of?

yakkomajuri commented:
@imbhargav5 Shouldn't really pose problems to the bots, but nevertheless by default it wouldn't lint .lock files and I've now added a prettier rule to ignore package.json too, given that and `yarn.lock` are usually what dependabot targets.

Moved packages to dependencies and sorted them too.

yakkomajuri commented:
Let me know if you want `11a0ddb` (test commit) reverted, I essentially just ran prettier on it.

imbhargav5 commented:
@yakkomajuri I am only thinking of this particular piece of code https://github.com/imbhargav5/rooks/blob/master/.github/workflows/version-bump.yml#L92. I don't want it to ever fail haha. If we can somehow configure lint-stage to not run in GitHub Actions env that would solve my paranoia. 

Also thank you for patiently addressing the fixes so far!

yakkomajuri commented:
Oh I mean if you really want I can set an env var on that workflow and check for it on the pre commit hook, but `prettier --write` doesn't fail (i.e. it's not like eslint), it just formats your files according to the spec. As of right now, it's configured to do that to files with the following extensions:

`html,js,ts,tsx,json,yml,css,scss`

yakkomajuri commented on .github/workflows/version-bump.yml:
```diff
@@ -2,18 +2,19 @@ name: Version bump
     env:
-      ALGOLIA_API_KEY:   ${{ secrets.ALGOLIA_API_KEY }}
+      ALGOLIA_API_KEY: ${{ secrets.ALGOLIA_API_KEY }}
+      IS_GH_ACTION: 1
```
here's the relevant diff

imbhargav5 commented:
Yeah. I wanted to be extra careful. Thanks for this @yakkomajuri !! Going to merge this.

imbhargav5 commented:
@all-contributors please add @yakkomajuri for code.
